src/sensor_state_data/data.py

```diff
@@ -279,3 +285,27 @@ def update_sensor(
             native_unit_of_measurement=native_unit_of_measurement,
             device_class=device_class,
         )
+
+    def fire_event(
+        self,
+        key: str,
+        event_type: None | str | int | float | date | datetime | Decimal,
+        event_subtype: None
+        | str
+        | int
+        | float
+        | date
+        | datetime
+        | Decimal
+        | None = None,
```

Comment:
I think it make sense to only allow `str` for `event_type` and `str | None = None` for event_subtype

Reply:
Changed